### PR TITLE
README update for v0.3.0 release

### PR DESCRIPTION
**Pull request by romanlutz**
Signed-off-by: Roman Lutz <rolutz@microsoft.com>

Based on https://github.com/fairlearn/fairlearn/pull/116 which was too complicated to rebase for DCO, so resubmitting this.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -21,28 +21,28 @@ For every pull request to `master` with automated tests you can check the logs o
...
+Reductions require an estimator to be passed that implements the `fit` method with the `sample_weight` argument. The constraints for reductions are all moments (see `fairlearn.reductions`) passed as instances of classes inheriting from `Moment`. Moments are vector functions that we use to formalize our constraints. The moment objects need to be passed as constraints to the constructor of a reduction technique, which internally uses the constraints to solve the optimization problem.
 
 ```python
 constraints = Moment()
-reduction = Reduction(estimator, objective=objective, constraints=constraints, use_predict_proba=False, **kwargs)
+reduction = Reduction(estimator, objective=objective, constraints=constraints, **kwargs)
 ```
 
-Reduction-based disparity mitigation algorithms (such as the ones under `fairlearn.reductions`) provide `fit`, `predict`, and `predict_proba` methods with the following signatures:
+Reduction-based disparity mitigation algorithms (such as the ones under `fairlearn.reductions`) provide `fit`, `predict`, and `_pmf_predict` methods with the following signatures:
```
Comment (riedgar-ms) [inline CONTRIBUTING.md]: Why are we mentioning `_pmf_predict` at all? Sorry if I've missed another discussion, but by putting it in the docs, we are sort of promising to support it.

File: README.md [hunk 2]
```diff
@@ -1,134 +1,142 @@
 [![Build Status](https://dev.azure.com/responsibleai/fairlearn/_apis/build/status/Nightly?branchName=master)](https://dev.azure.com/responsibleai/fairlearn/_build/latest?definitionId=23&branchName=master) ![MIT license](https://img.shields.io/badge/License-MIT-blue.svg) ![PyPI](https://img.shields.io/pypi/v/fairlearn?color=blue)
```
Comment (riedgar-ms) [inline README.md]: So there will be one final update once this is in the release branch, to get the branchName correct

File: README.md [hunk 3]
```diff
@@ -1,134 +1,142 @@
...
+
+or you can clone the repository locally via
 
-DP refers to Demographic Parity, EO to Equalized Odds, and BGL to Bounded Group Loss. For more information on these and other terms we use in this repository please refer to the [terminology page](TERMINOLOGY.md).
+```python
+git clone git@github.com:fairlearn/fairlearn.git
+```
 
-To request additional algorithms or fairness definitions, please open a new issue.
+To verify that it works run
```
Comment (riedgar-ms) [inline README.md]: We should clarify that 'it' here refers to the clone, not the pip install

Comment (riedgar-ms) [reply README.md]: I've fixed this.

File: CONTRIBUTING.md (hunk 1 above)
Comment (romanlutz) [reply CONTRIBUTING.md]: It had predict_proba there before, so I just went and replaced it. Better be consistent with the code than mention stuff we don't even have. If you want to remove it I am not opposed to that.

Comment (riedgar-ms) [reply CONTRIBUTING.md]: Yes - I was surprised it hadn't been removed entirely, and was wondering if I'd missed a discussion

**Pull request merged**